src/Agent.Listener/SelfUpdater.cs

```diff
@@ -179,7 +188,7 @@ private async Task DownloadLatestAgent(CancellationToken token)
             try
             {
                 // Download the agent, using multiple attempts in order to be resilient against any networking/CDN issues
-                for (int attempt = 1; attempt <= Constants.AgentDownloadRetryMaxAttempts; attempt++)
+                for (int attempt = 1; attempt <= Constants.AgentDownloadRetryMaxAttempts && !downloadSucceeded; attempt++)
```

Comment:
Is this check somehow related to package checksum validation? Could this break current package download behavior?

Reply:
Self Updater uses multiple attempts to download the agent package — via the loop. When the download succeeded, there is an exit from the loop. We can use the `break;` option for it, or add the related condition — exit the loop when the attempts run out or when the download completes successfully. So this does not break the current behavior. One condition instead of many breaks.